privacy_partner_report/controllers/main.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+import json
+from openerp import http
+
+from openerp.addons.web.controllers.main import serialize_exception
+from openerp.addons.web.controllers.main import Reports
+
+
+class ReportsExtended(Reports):
+    # HACK of https://github.com/odoo/odoo/pull/24964
+
+    @http.route()
+    @serialize_exception
+    def index(self, action, token):
+        try:
+            action = json.loads(action)
+            if "data" in action.keys():
+                action["datas"] = action["data"]
+            action = json.dumps(action)
+        except KeyError:
```

Comment:
This cannot happen now. I think it would have been simplest to keep it as it was and just use a different variable name instead of using the same `action` for both the dict and the string. Something like:

        try:
            action_dict = json.loads(action)
            action_dict["datas"] = action_dict["data"]
            action = json.dumps(action_dict)
        except KeyError:
            pass

Otherwise, keep it as it is and remove the now impossible try-except block.